### PR TITLE
[KernelGen][Nvidia] Add miopen_batch_norm_backward operator with Triton kernel

### PR DESCRIPTION
**Pull request by Dingxingdi**
## Summary
Adds a Triton kernel implementation for `miopen_batch_norm_backward`.

## Testing
- Correctness: `CUDA_VISIBLE_DEVICES=0 python -m pytest tests/test_miopen_batch_norm_backward.py -x -v --timeout=60`
- Benchmark: `CUDA_VISIBLE_DEVICES=0 python -m pytest benchmark/test_miopen_batch_norm_backward.py --level core -s`
- Tested on: NVIDIA H20, Docker container `dxd`, PyTorch CUDA runtime 12.9, driver 595.45.04

## Performance
NVIDIA H20, `--level core`, 15 cases total, arithmetic mean speedup: **2.336x**.

### float16
| Shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---:|---:|---:|
| [4, 16, 64, 4] | 0.012 | 0.012 | 1.022 |
| [16, 16, 8, 48] | 0.022 | 0.013 | 1.706 |
| [16, 16, 8, 88] | 0.029 | 0.015 | 1.961 |
| [16, 16, 128] | 0.014 | 0.012 | 1.199 |
| [20, 6, 65536] | 1.868 | 0.284 | 6.573 |

### float32
| Shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---:|---:|---:|
| [4, 16, 64, 4] | 0.011 | 0.012 | 0.886 |
| [16, 16, 8, 48] | 0.020 | 0.013 | 1.547 |
| [16, 16, 8, 88] | 0.029 | 0.015 | 1.921 |
| [16, 16, 128] | 0.013 | 0.012 | 1.074 |
| [20, 6, 65536] | 2.170 | 0.414 | 5.241 |

### bfloat16
| Shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---:|---:|---:|
| [4, 16, 64, 4] | 0.013 | 0.014 | 0.951 |
| [16, 16, 8, 48] | 0.022 | 0.013 | 1.667 |
| [16, 16, 8, 88] | 0.029 | 0.014 | 2.036 |
| [16, 16, 128] | 0.014 | 0.012 | 1.230 |
| [20, 6, 65536] | 1.871 | 0.310 | 6.031 |

## Multi-backend Testing
| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---:|---|
| NVIDIA H20 | PASS | PASS (15 cases, --level core) | 2.336 | Primary |
| Tianshu | FAIL | PASS (60 cases) | 6.044 | Existing backend summary reports failure |
| Muxi | FAIL | FAIL | — | Existing backend summary reports failure |
| Ascend | FAIL | N/A | — | Existing backend summary reports failure |
| Hygon | PASS | PASS (60 cases) | 2.454 | — |

## Files Changed
- `src/flag_gems/ops/miopen_batch_norm_backward.py`: Triton kernel implementation
- `tests/test_miopen_batch_norm_backward.py`: accuracy test
- `benchmark/test_miopen_batch_norm_backward.py`: performance benchmark
- `src/flag_gems/ops/__init__.py`: register import and `__all__`
- `src/flag_gems/__init__.py`: register `_FULL_CONFIG`
- `conf/operators.yaml`: add operator entry with `alpha: '5.4'`

### CODE REVIEW

File: tests/test_miopen_batch_norm_backward.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    if affine:
+        output_mask = [True, True, True]
+    else:
+        output_mask = [True, False, False]
+
+    (
+        ref_in_grad,
+        ref_weight_grad,
+        ref_bias_grad,
+    ) = torch.ops.aten.native_batch_norm_backward(
```
Comment (w1120029931-bit) [inline tests/test_miopen_batch_norm_backward.py:66]: Why is the comparison for adding the miopen_batch_norm_backward operator made with torch.ops.aten.native_batch_norm_backward?

Comment (Dingxingdi) [reply tests/test_miopen_batch_norm_backward.py:66]: Because aten::miopen_batch_norm_backward is backend-specific and cannot be executed as the reference path on NVIDIA, while native_batch_norm_backward provides the same batch norm backward math.

However, I found another issue. The MIOpen forward path passes the saved save_var directly to MIOpen backward, so in the native reference it should correspond to save_invstd directly, not rsqrt(save_var + eps).

I fixed the issue.

**Review by bin913 — APPROVED** [review]
